### PR TITLE
Document import library

### LINKED ISSUE #483: STL: We should better tag those files which might end up in import libs
In the review https://github.com/microsoft/STL/pull/457 @StephanTLavavej asked for the "ominous comment"

https://github.com/microsoft/STL/blob/b3504262fe51b28ca270aa2e05146984ef758428/stl/inc/xfacet#L19-L22

to be added to a ["core" header](https://github.com/microsoft/STL/wiki/What's-the-difference-between-(core)-and-non-(core)-headers%3F) on the basis that it was included in an import lib. However, that would be incongruous with other core headers, which we already include in our import libs.

We should remove the "ominous comment" from headers and put it in that wiki page, add a link from each "core" header to the wiki page, and ensure the wiki page explains the same content as the "ominous comment". We should also ensure that each .cpp file in the import lib has an ominous comment.

We should probably also update the ominous comment to not talk about the length of included stuff, similar to the following:

```
// This file is part of the msvcprt.lib and msvcprtd.lib import libraries.
// Therefore, it must not depend on any settings that affect layout or behavior of standard
// library types like _CONTAINER_DEBUG_LEVEL or _ITERATOR_DEBUG_LEVEL.
// In particular, basic_string must not be included here.
```

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Resolve #483

### CODE REVIEW

File: docs/import_library.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+The STL import library is called msvcprt.lib for release, msvcprtd.lib for debug,
+and it links to corresponding msvcp140.dll and msvcp140d.dll DLLs.
+
+However, the STL import library defines some functions and variables on its own.
+This allows to:
+ * Extend the STL implementation without altering the DLL export surface;
+ * Efficiently and conveniently export variables and constants.
+
+The caveats of this technic are:
```
Comment (bhardwajs) [inline docs/import_library.txt:18]: Should this be `technique`?

Comment (AlexGuteniev) [reply docs/import_library.txt:18]: fixed

**Review by StephanTLavavej — APPROVED** [review]

File: stl/inc/regex [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <cwchar>
+#include <initializer_list>
```
[the remark this answers is not in the document]
Comment (AlexGuteniev) [reply stl/inc/regex]: The intent was to remove `<initializer_list>` from import library header files.
If we don't do this, we should put a scary comment there instead too.

[the remark this answers is not in the document]
Comment (StephanTLavavej) [reply stl/inc/regex]: As I mentioned on Discord, it's core, so it doesn't need a comment:

https://github.com/microsoft/STL/blob/9231abe46d466a0f262db234fd3bd4de1170ee45/stl/inc/initializer_list#L1

[the remark this answers is not in the document]
Comment (frederick-vs-ja) [reply stl/inc/regex]: > I'd like to drop these structural changes, keeping only the documentation changes. If there's a reason to change the include structure, we can do so in a separate PR.

Yep. I'll try to do the restructure for `<xstddef>`. After looking into it for about 1 hour, I think most or all of its contents should be moved to other headers for the sake of throughput.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving the documentation for this confusing area and upgrading `<xcharconv_tables.h>`! :scroll: :heart_eyes_cat: :white_check_mark:

**Pull request merged**